src/Compilers/CSharp/Portable/Syntax/NameColonSyntax.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.CodeAnalysis.CSharp.Syntax
+{
+    partial class NameColonSyntax
+    {
+        public override ExpressionSyntax Expression => Name;
+
+        internal override BaseExpressionColonSyntax WithExpressionCore(ExpressionSyntax expression)
+        {
+            return WithName((IdentifierNameSyntax)expression);
```

Comment:
Should we do this instead?
```suggestion
            if (expression is IdentifierNameSyntax identifierName)
                return WithName(identifierName);
            return SyntaxFactory.ExpressionColon(expression, this.ColonToken);
```

Reply:
@CyrusNajmabadi Could this be problematic? We don't have a precedent of doing something like this so far.